### PR TITLE
[MM-46054] Log missing default channels

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
A weird code fragment in `channel.go` caused MM to call `errors.As` on a `(*model.AppError)(nil)` which lead to a crash in the `Unwrap` method. The code effectively only skipped missing channels and did nothing useful. This PR removes the code and replaces it with a log statement.

#### Ticket Link
[MM-46054](https://mattermost.atlassian.net/browse/MM-46054)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Lovely

File: app/channel.go [hunk 1]
```diff
@@ -91,17 +91,10 @@ func (a *App) JoinDefaultChannels(c request.CTX, teamID string, user *model.User
 		}
 	}
 
-	var err *model.AppError
```
Comment (isacikgoz) [inline app/channel.go:94]: Wondering how this didn't caught by the compiler or linter as the err is not used anywhere 🤔

**Review by isacikgoz — APPROVED** [review]

File: app/channel.go (hunk 1 above)
Comment (noxer) [reply app/channel.go:94]: It war read in line 100 and written in the lines 101 and 103, that's all it needs to be considered a useful member of ~society~ code by the compiler.

**Comment by Phrynobatrachus** [comment]
Manually triggered another e2e run as the first had some issue pulling the docker tag.

https://automation-dashboard.vercel.app/cycles/a49c7ac7-95de-4fcd-80da-a93800f3478d

**Pull request merged**